hedera-node/src/main/java/com/hedera/services/queries/QueriesModule.java

```diff
@@ -60,21 +59,24 @@ public static AnswerFlow provideAnswerFlow(
 			final PlatformSubmissionManager submissionManager,
 			@HapiThrottle final FunctionalityThrottling hapiThrottling
 	) {
-		if (nodeType == STAKED_NODE) {
-			return new StakedAnswerFlow(
-					fees,
-					accountNums,
-					stateViews,
-					usagePrices,
-					hapiThrottling,
-					submissionManager,
-					queryHeaderValidity,
-					transactionPrecheck,
-					hapiOpPermissions,
-					queryFeeCheck);
-		} else {
-			return new ZeroStakeAnswerFlow(queryHeaderValidity, stateViews, hapiThrottling);
-		}
+		final var stakedFlow = new StakedAnswerFlow(
+				fees,
+				accountNums,
+				stateViews,
+				usagePrices,
+				hapiThrottling,
+				submissionManager,
+				queryHeaderValidity,
+				transactionPrecheck,
+				hapiOpPermissions,
+				queryFeeCheck);
+
+		final var zeroStakeFlow = new ZeroStakeAnswerFlow(
+				queryHeaderValidity,
+				stateViews,
+				hapiThrottling);
+
+		return new StakeAwareAnswerFlow(nodeInfo, stakedFlow, zeroStakeFlow);
```

Comment:
Can we check `nodeInfo.isSelfZeroStake()` here and return needed object, since wont need both objects at same time ?

Reply:
If we call `nodeInfo.isSelfZeroStake()` here, it will cause a NPE because the working `StateAccessor` will not yet have had its children updated [here](https://github.com/hashgraph/hedera-services/blob/02019-M-TrackReferencedEntities/hedera-node/src/main/java/com/hedera/services/ServicesState.java#L365).

Reply:
I see. Thank you!